tests/test_html.py

```diff
@@ -959,13 +962,18 @@ def test_no_alternate_url_json(app_client, path):
     (
         (
             "/fivethirtyeight/twitter-ratio%2Fsenators",
-            "/fivethirtyeight/twitter-2Dratio-2Fsenators",
+            "/fivethirtyeight/twitter-ratio~2Fsenators",
         ),
+        # TODO: re-enable this test
+        # (
+        #    "/fixtures/table%2Fwith%2Fslashes.csv",
+        #    "/fixtures/table~2Fwith~2Fslashes~2Ecsv",
+        # ),
```

Comment:
The problem can be replicated by hitting: http://127.0.0.1:8001/fixtures/table%2Fwith%2Fslashes.csv

Reply:
This is happening because of this code in the `csv` handler, which rewrites the `scope` for some reason:

https://github.com/simonw/datasette/blob/5ce259aa4b35e69e52c5bad758542d3ebaf4f449/datasette/views/base.py#L285-L303

Reply:
Solution here is to create a brand new replace `Request` rather than modifying the existing one in place:
```diff
diff --git a/datasette/views/base.py b/datasette/views/base.py
index 8f5a385..1c0c3f9 100644
--- a/datasette/views/base.py
+++ b/datasette/views/base.py
@@ -10,6 +10,7 @@ import pint
 
 from datasette import __version__
 from datasette.database import QueryInterrupted
+from datasette.utils.asgi import Request
 from datasette.utils import (
     add_cors_headers,
     await_me_maybe,
@@ -291,6 +292,7 @@ class DataView(BaseView):
             if not request.args.get(key)
         ]
         if extra_parameters:
+            # Replace request object with a new one with modified scope
             if not request.query_string:
                 new_query_string = "&".join(extra_parameters)
             else:
@@ -300,7 +302,8 @@ class DataView(BaseView):
             new_scope = dict(
                 request.scope, query_string=new_query_string.encode("latin-1")
             )
-            request.scope = new_scope
+            receive = request.receive
+            request = Request(new_scope, receive)
         if stream:
             # Some quick soundness checks
             if not self.ds.setting("allow_csv_stream"):
```